tools/build_rules/verifier_test/cc_indexer_test.bzl

```diff
@@ -180,22 +180,29 @@ def _transitive_entries(deps):
             compressed += dep[KytheEntries].compressed
     return KytheEntries(compressed = depset(transitive = compressed), files = depset(transitive = files))
 
+def _fix_path_for_generated_file(path):
```

Comment:
Could you explain why you need to do this? My concern is that we will also need to change the "real" build implementation to do something similar.

Reply:
The initial problem is that when building the command line parameters for the protoc compiler, the call to paths._relativize fails with the following error:

Path 'bazel-out/k8-fastbuild/bin/kythe/testdata/indexers/proto/_virtual_imports/testdata5_proto/kythe/testdata/indexers/proto/gen/testdata5.gen.proto' is not beneath 'kythe/testdata/indexers/proto'.

If we manage to get past this error, the next problem we run into is in the protoc compiler itself.  When constructing the GeneratedSourceInfo message used to generate the metadata, the protoc compiler uses  path() in the FileDescriptor for the proto being compiled, in this case testdata5.gen.proto in the _virtual_imports directory, which results in metadata that references a file with _virtual_imports in the path. This becomes a problem downstream because the proto paths in the cxx index don't match the paths in the proto index, so we lose the "generates" relationship. 

By stripping everything up to "kythe/testdata/..." we get past the relativize issue, and more importantly the paths in the proto index and cxx index match, all we have to do now is make sure the vnames that are constructed have the correct root that resolve to the generated files directory.  For that we use the .gen.proto file extension.

A similar issue with _virtual_imports was identified and fixed in https://github.com/bazelbuild/bazel/commit/f65cc726655eb81c6e0aeb04d05b42831640dce9.